python/grass/temporal/register.py

```diff
@@ -31,6 +31,44 @@
 ###############################################################################
 
 
+def build_abstract_map_dataset(map_item, element):
+    """Build AbstractMapDataset from map item and element type.
+
+    :param map_item: The id of the map to build an abstract dataset for.
+                     Can be of the form: mapname, mapname@mapsetname, or
+                     mapname:layer / mapname:layer@mapsetname for vector
+                     maps
+    :param element: The mapset element of the maps: cell, grid3d or vector
+    :return: ID String build with the AbstractMapDataset class
+
+    .. code-block:: python
+
+        >>> import grass.temporal as tgis
+        >>> tgis.register.build_abstract_map_dataset("vectormap:1@PERMANENT", "vector")
+        'vectormap:1@PERMANENT'
+        >>> tgis.register.build_abstract_map_dataset("rastermap@PERMANENT", "cell")
+        'rastermap@PERMANENT'
+        >>> tgis.register.build_abstract_map_dataset("raster3dmap@PERMANENT", "grid3d")
+        'raster3dmap@PERMANENT'
+
+    """
+    map_name, map_mapset = (
+        map_item.split("@")[0:2] if "@" in map_item else (map_item, None)
+    )
+    map_name, map_layer = map_name.split(":") if ":" in map_name else (map_name, None)
+    if not map_mapset:
+        result = gscript.find_file(element=element, name=map_name)
+        if result["mapset"]:
+            map_mapset = result["mapset"]
+        else:
+            gscript.fatal(
+                _("{type} map <{map_name}> not found on search path").format(
+                    type=type, map_name=map_name
+                )
+            )
```

Comment:
This is not in the original code. The requirement for map existence needs to be stated in the doc.

Reply:
I assume you mean in the doc string of the function?!

That said, the function does throw an error if the map is not found, only later in the code (see: https://github.com/OSGeo/grass/blob/b5c478d714077e31a5a481bb52fac78f10ec08d9/python/grass/temporal/register.py#L295-L299)

Here is the current behavior:
```
t.create output=strds type=strds title=test description=test
t.register input=strds maps=map_does_not_exist type=raster start="2001-01-01" end="2001-01-01"
ERROR: Unable to update raster map <map_does_not_exist@m2>. The map does
not exist.
```
So this addition, basically makes the register function fail earlier...

Reply:
So, perhaps the new function is existing_map_to_id() or something like that. In C, I sometime end up with x_parameter_to_y_or_fail just to make it clear that fatal/exit will be called.

Will the check later still make sense then? Maybe just check for failure (e.g. try-except) instead of checking ahead there. This is lower priority, though.

Yes, I mean the docstring.

Reply:
Solved in the `build_id_from_search_path` function and the documentation is updated accordingly.